### PR TITLE
Organize test/modules

### PR DESCRIPTION
**Pull request by ben-albrecht**
This PR partially implements #5599 and splits the module tests under `test/modules/{standard,packages}` into `test/library/{standard,packages}`.

The new hierarchy looks like this:

```
test
├── library
│   ├── standard
│   │   ├── Assert
│   │   ├── Barrier
│   │   ├── ...
│   ├── packages
│   │   ├── BLAS
│   │   ├── Curl
│   │   ├── ...
├── modules
│   ├── ... (unchanged)
```

I have opted to not organize the remaining `modules/` test in this PR, given the number of tests (230). This seems better fit for a separate PR (if at all).

Other changes include:

- Updated test directory names to reflect module names exactly (including case-sensitivity)
    - e.g. `blas` -> `BLAS`, `linearalgebra` -> `LinearAlgebra`
- Updated references of `test/modules` to `test/library` when referring to `standard/` or `packages/`
- Moved `test/link` -> `test/compflags`

**Testing**

- [x] `start_test test/library/`
- [x] `make docs`
   - [x] Confirm docs look fine
- [x] `paratest`

### CODE REVIEW

File: test/library/packages/LinearAlgebra/correctness/lp.chpl [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/* LAPACK tests */
```
Comment (benharsh) [inline test/library/packages/LinearAlgebra/correctness/lp.chpl]: Did you mistakenly add this file and a binary named ``lp`` ?

File: test/library/packages/TOML/BurntSushi/toml2json.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
```
Comment (benharsh) [inline test/library/packages/TOML/BurntSushi/toml2json.py]: Should this file be included in this PR?

**Comment by bradcray** [comment]
I haven't caught up on the full conversation that led to this change, but am wondering whether anyone proposed changing $CHPL_HOME/modules to `$CHPL_HOME/library` (or `libraries`) or `$CHPL_HOME/lib` for greater symmetry and to say more about what the `modules` library does rather than what it contains.

**Comment by ben-albrecht** [comment]
> I haven't caught up on the full conversation that led to this change

Responding back in #5599 with the original context.

**Pull request merged**